pkg/search/proxy/controller.go

```diff
@@ -312,3 +319,24 @@ func dynamicClientForClusterFunc(clusterLister clusterlisters.ClusterLister,
 		return dynamic.NewForConfig(clusterConfig)
 	}
 }
+
+func (ctl *Controller) storageReadinessCheck() bool {
+	return ctl.store.ReadinessCheck() == nil
+}
+
+// Hook waits for the controller to be in a storage ready state.
```

Comment:
```suggestion
// Hook waits for the controller to be in a storage ready state.
// Here, even if the initialization is not completed within the timeout interval,
// nil is still returned because the cache is per-type and per-cluster layer,
// and we want to avoid making the whole component as not ready, if the request for
// one of the resource types requires reinitialization, requests for all other
// resource types can still be handled properly.
```